### PR TITLE
Feature/skeleton view

### PR DESCRIPTION
**Pull request by GSKang94**

### CODE REVIEW

File: RickAndMorty-iOS/Views/Shared/EmptyData.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    animation.isRemovedOnCompletion = false
+
+    gradient.add(animation, forKey: "shimmer")
+
+    currentCell.contentView.layer.addSublayer(gradient)
+    shimmerLayer = gradient
+}
+
+func hideLoadingAnimation(currentCell: UICollectionViewCell) {
+    shimmerLayer?.removeFromSuperlayer()
```
Comment (CalvinPak) [inline RickAndMorty-iOS/Views/Shared/EmptyData.swift]: What if you change this line to `currentCell.contentView.layer.sublayers?.removeAll()` ?

File: RickAndMorty-iOS/Views/Episodes/EpisodesViewController.swift [hunk 2]
```diff
@@ -62,10 +72,32 @@ class EpisodesViewController: UIViewController {
...
+            }
+
+            let episode = data as? RickAndMortyAPI.GetEpisodesQuery.Data.Episodes.Result
+            let cell = collectionView.dequeueConfiguredReusableCell(using: self.episodesView.episodeCell,
+                                                                    for: indexPath,
+                                                                    item: episode)
+            cell.upperLabel.text = episode?.name
+            cell.lowerLeftLabel.text = episode?.episode
+            cell.lowerRightLabel.text = episode?.air_date
+
```
Comment (CalvinPak) [inline RickAndMorty-iOS/Views/Episodes/EpisodesViewController.swift:91]: and put `hideLoadingAnimation(currentCell: cell)` on this line

File: RickAndMorty-iOS/Views/Episodes/EpisodesViewController.swift [hunk 3]
```diff
@@ -62,10 +72,32 @@ class EpisodesViewController: UIViewController {
...
-        dataSource = DataSource(collectionView: episodesView.collectionView, cellProvider: { (collectionView, indexPath, episode) -> UICollectionViewCell? in
-            return collectionView.dequeueConfiguredReusableCell(using: self.episodesView.episodeCell,
-                                                                for: indexPath,
-                                                                item: episode)
+        dataSource = DataSource(collectionView: episodesView.collectionView, cellProvider: { (collectionView, indexPath, data) -> UICollectionViewCell? in
+
+            // section with empty episode cells
+            if indexPath.section == 1 {
+                let cell = collectionView.dequeueConfiguredReusableCell(using: self.episodesView.episodeCell, for: indexPath, item: data as? EmptyData )
+//                showLoadingAnimation(currentCell: cell)
```
Comment (CalvinPak) [inline RickAndMorty-iOS/Views/Episodes/EpisodesViewController.swift]: and uncomment this line `showLoadingAnimation(currentCell: cell)` ?

**Review by CalvinPak — APPROVED** [review]
Amazing. 

For the animation, have you tried to remove the animation layer using `currentCell.contentView.layer.sublayers?.removeAll()` (or `removeLast()` / `removeFirst()`)?

File: RickAndMorty-iOS/Views/Shared/EmptyData.swift (hunk 1 above)
Comment (GSKang94) [reply RickAndMorty-iOS/Views/Shared/EmptyData.swift]: That worked like charm!!
Thank you

**Pull request merged**